File: TDT4120/datastrukturer/stack.py
class Stack:
    stackList = []

    def push(self, element):
        self.stackList.append(element)

    def pop(self):
        try:
            return self.stackList.pop()
        except:
            #print underflow
            return None
        
    def peek(self):
        try:
            return self.stackList[-1]
        except:
            return None


#test
'''
s = Stack()
s.push('a')
s.push('b')
print s.peek()
print s.pop()
'''

File: TDT4100/src/øving2/Stack.java
package øving2;

import java.util.ArrayList;

public class Stack {
	
	ArrayList<String> stringList;
	
	
	public Stack() {
		stringList = new ArrayList<String>();
	}
	
	
	public void push(String str) {
		stringList.add(0, str);
	}
	
	
	public String pop() {
		if (stringList.isEmpty()) {
			return null;
		} else {
			return stringList.remove(0);
		}
	}
	
	
	public String peek(int n) {
		if (n < 0 || n > stringList.size() - 1) {
			return null;
		} else {
			return stringList.get(n);
		}
	}
	
	
	public 	int getSize() {
		return stringList.size();
	}
	
	
	public String toString() {
		return "[Stack " + "size: " + getSize() + "]";
	}

}
